target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8074-rax120v2.dts

```diff
@@ -0,0 +1,522 @@
+#include "ipq8074-ess.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "Netgear RAX120v2";
+	compatible = "netgear,rax120v2", "qcom,ipq8074";
+
+	aliases {
+		serial0 = &blsp1_uart5;
+
+		led-running = &led_system_white;
+		led-upgrade = &led_system_white;
+		led-internet = &led_wan_white;
+
+		label-mac-device = &dp1;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " ubi.mtd=rootfs root=/dev/ubiblock0_0";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		rfkill {
+			label = "rfkill";
+			gpios = <&tlmm 25 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RFKILL>;
+		};
+
+        wps {
+			label = "wps";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+        reset {
+            label = "reset";
+			gpios = <&tlmm 54 GPIO_ACTIVE_LOW>;
+            linux,code = <KEY_RESTART>;
+        };
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		sck-gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+		mosi-gpios = <&tlmm 19 GPIO_ACTIVE_HIGH>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <2>;
+			enable-gpios = <&tlmm 20 GPIO_ACTIVE_HIGH>;
+			spi-max-frequency = <1000000>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_system_white: system-white {
+			label = "white:system";
+			gpios = <&led_gpio 0 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+
+		led_24g_white {
+			label = "white:24g";
+			gpios = <&led_gpio 1 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		led_5g_white {
+			label = "white:5g";
+			gpios = <&led_gpio 2 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		led_usb1_white {
+			label = "white:usb1";
+			gpios = <&led_gpio 3 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+
+		led_usb2_white {
+			label = "white:usb2";
+			gpios = <&led_gpio 4 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+
+		led_wan_white: wan-white {
+			label = "white:wan";
+			gpios = <&led_gpio 5 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+
+		led_aqr_green {
+			label = "green:aqr";
+			gpios = <&led_gpio 6 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_GREEN>;
+		};
+
+		led_aqr_red {
+			label = "red:aqr";
+			gpios = <&led_gpio 10 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_RED>;
+		};
+
+		led_aqr_white {
+			label = "white:aqr";
+			gpios = <&led_gpio 11 GPIO_ACTIVE_LOW>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+
+		led_wps_white {
+			label = "white:wps";
+			gpios = <&tlmm 40 GPIO_ACTIVE_HIGH>;
+			color = <LED_COLOR_ID_WHITE>;
+		};
+	};
+};
+
+&clocks {
+	g762_clk: fixedclk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+	};
```

Comment:
It's the clock for the fan and in the source code from Netgear, in file `git_home/linux.git/arch/arm64/boot/dts/qcom/qcom-ipq807x-soc.dtsi` I find:

```
1066             fan_left: g761@3e {
1067                 compatible = "gmt,g761";
1068                 reg = <0x3e>;
1069                 clocks =<&g761_clk>;
1070                 fan_gear_mode = <0>;
1071                 fan_starty = <1>;
1072                 pwm_polarity = <0>;
1073             };
...
2118         g761_clk: g761-oscillator{
2119               compatible = "fixed-clock";
2120               #clock-cells = <0>;
2121               clock-frequency = <32768>;
2122
2123         };
```

@jewwest used the same values but 'compatible = "gmt,g762"; instead of "compatible = "gmt,g761"; like in the Netgear sources.

I never heard the fan running so not sure if 'g762' is correct or if the fan code is already part of the soc code.

Reply:
I will use the SOC's sleep clock and see how I can verify that it work.

I really thought that the vendor DTS is a hint. But since I (almost certainly) converted my first RAX120v2 into a paper weight I can look at the IC on the PCP.

Reply:
Good Morning @robimarko, I had a look at the board and right next to the fan connector is a 'W51S G761' IC.

![Netgear-RAX120v2-fan](https://user-images.githubusercontent.com/52218/236743221-2934290c-7bcb-4df3-b6fe-a7c096e9f4e9.jpeg)

Reply:
You can't really trust the vendor's DTS for anything, they will just go the path of least resistance no matter how broken it is.

If it's G761 then `gmt,g763` should be used as that seems to be the single-channel as well, though kernel driver makes no difference between them.

Reply:
Ok, but do I keep the fan section and refer to the sleep clock? And how do I refer to the sleep clock?

right now:

```
296 &blsp1_i2c2 {
297     status = "okay";
298
299     fan: g762@3e {
300         compatible = "gmt,g762";
301         reg = <0x3e>;
302         clocks =<&g762_clk>;
303         fan_gear_mode = <0>;
304         fan_start = <1>;
305         pwm_polarity = <0>;
306     };
307 };
```

Reply:
Yes obviously, but drop the fan label if that is not being referenced on and for the clock its just:
`clocks =<&sleep_clk>;`